zcash_primitives/src/block.rs

```diff
@@ -1,8 +1,18 @@
+use hex;
+use std::fmt;
 use std::ops::Deref;
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub struct BlockHash(pub [u8; 32]);
 
+impl fmt::Display for BlockHash {
```

Comment:
Aside from the name, `BlockHash` is the same as  `TxId` and in this commit we write the same `fn fmt` twice. This makes me wonder if there should be a single type (named `ByteArray32` or something like that) rather than two.

Reply:
I'm not convinced that the extra indirection to the underlying bytes is worth it yet. If we find the reverse-hex format is bleeding in too much from `zcashd` then we can revisit this.